apps/session-manager/src/controllers/authentication.ts

```diff
@@ -658,3 +669,64 @@ export const acs: (
       E.toUnion,
     );
   };
+
+export const acsTest: (
+  userPayload: unknown,
+) => (
+  dependencies: AcsDependencies & { clientProfileRedirectionUrl: string },
+) => TE.TaskEither<
+  Error,
+  | IResponseErrorInternal
```

Comment:
as done with other endpoints, we can remove `IResponseErrorInternal` because `toExpressHandler` utility already transform every `Error` instance to an `IResponseErrorInternal`. but we would need to extend the if with 
```ts
} else if (acsResponse.kind === "IResponseErrorInternal") {
      throw Error(acsResponse.detail);
    }
```
what do you think?

Reply:
We should remap the left from the pipe and the response from the `acs`. This will make the code less linear only to handle the internal error at handler level. I prefer this version. When we refactor the acs returning a taskEither instead of a promise we can consider this refactory